backend/src/controllers/items-controller.ts

```diff
@@ -0,0 +1,136 @@
+import { Request, Response } from 'express';
+import { validationResult } from 'express-validator';
+import Item from '../models/item';
+import User from '../models/user'
+import InventoryItem from "../models/inventory-item";
+import mongoose from 'mongoose';
+import {ObjectId} from "mongodb";
+
+//Creates an item
+export const createItem = async (req: Request, res: Response) => {
+    //Validation check
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        console.log("errors", errors);
+        return res.status(400).json({errors: errors.array()});
+    }
+    //Duplicated item check
+    const dupeItem = Item.findOne({name: req.body.name})
+    if (!dupeItem) {
+        return res.status(409).json({error: "An item with the same name already exists."});
+    }
+    //Checks if the ID matches the owner of the item
+    const itemOwnerId = req.params._id;
+    const itemOwnerExistence = User.findOne({_id: itemOwnerId});
+    if (!itemOwnerExistence) {
+        return res.status(404).json({error: "This id does not match the owner of the item."});
+    }
+    const name = req.body.name as string;
+    const description = req.body.description as string;
+    const pickupLocation = req.body.pickupLocation as string;
+    const pickupTime = req.body.pickupTime as string;
+
+
+
+    //Interface for a new item
+    const item = new Item({
+        name: name,
+        description: description,
+        pickupLocation: pickupLocation,
+        pickupTime: pickupTime,
+        itemOwnerId: itemOwnerId
```

Comment:
Populate this field with `itemOwnerExistence._id` instead of `itemOwnerId` so you store the `mongodb`  `ObjectId` and not a string of it